src/interactivity/Interactivity.js

```diff
@@ -128,41 +128,54 @@ export default class Interactivity {
         this._prevHoverFeatures = [];
         this._prevClickFeatures = [];
         this._numListeners = {};
+        this._isAutoChangePointerEnabled = options.autoChangePointer;
 
         const allLayersReadyPromises = layerList.map(layer => layer._context);
         return Promise.all(allLayersReadyPromises)
             .then(() => {
                 postCheckLayerList(layerList);
+                this._map = layerList[0].map;
+
                 this._subscribeToLayerEvents(layerList);
-                this._subscribeToMapEvents(layerList[0].map);
-            }).then(() => {
-                if (options.autoChangePointer) {
+                this._subscribeToMapEvents(this._map);
+
+                if (this._isAutoChangePointerEnabled) {
                     this._setInteractiveCursor();
                 }
             });
     }
 
     _setInteractiveCursor () {
-        const map = this._layerList[0].map; // All layers belong to the same map
-        if (!map.__carto_interactivities) {
-            map.__carto_interactivities = new Set();
+        if (!this._map.__carto_interactivities) {
+            this._map.__carto_interactivities = new Set();
         }
-        this.on('featureHover', event => {
-            if (event.features.length) {
-                map.__carto_interactivities.add(this);
-            } else {
-                map.__carto_interactivities.delete(this);
-            }
-            map.getCanvas().style.cursor = (map.__carto_interactivities.size > 0) ? 'pointer' : '';
-        });
+        this.on('featureHover', event => this._onFeatureHover(event, this._map));
+    }
+
+    _onFeatureHover (event, map) {
+        if (event.features.length) {
+            map.__carto_interactivities.add(this);
+        } else {
+            map.__carto_interactivities.delete(this);
+        }
+        map.getCanvas().style.cursor = (map.__carto_interactivities.size > 0) ? 'pointer' : '';
     }
 
     _subscribeToMapEvents (map) {
-        map.on('mousemove', this._onMouseMove.bind(this));
-        map.on('click', this._onClick.bind(this));
+        this._onMouseMoveBound = this._onMouseMove.bind(this);
+        map.on('mousemove', this._onMouseMoveBound);
+
+        this._onClickBound = this._onClick.bind(this);
+        map.on('click', this._onClickBound);
+
         this._disableWhileMovingMap(map);
     }
 
+    _unsubscribeToMapEvents (map) {
```

Comment:
maybe a rename to `_unsubscribeFromMapEvents` here?

Reply:
I had the same doubt the other day, and I searched for it 😂 
StackExchange says this: https://english.stackexchange.com/questions/102994/unsubscribe-to-vs-unsubscribe-from. (Take into account that sanctioned is a false friend here).

So, do you prefer `_unsubscribeFromMapEvents`?

Reply:
My ear was used to 'from', but it's a good thing to know that 'to' works fine too. So let's keep that as it is (and keep learning good ol' english!) 😄